[BUG] Runtime Panic

**Describe the bug**
I was able to crash the `wasmcloud` runtime while testing an actor locally. 

**To Reproduce**
Steps to reproduce the behavior:
1. I used the go actor in this PR https://github.com/wasmCloud/examples/pull/43
2. make && make start (as long as your actor key is correct in the manifest)
3. `curl 'localhost:8080/'` -> Malformed response 
4. `curl 'localhost:8080/'` -> Bus error: 10 (from wasmcloud log)

**Expected behavior**
Runtime not crash

**Screenshots**
![Screen Shot 2021-04-23 at 11 31 33 AM](https://user-images.githubusercontent.com/15827604/115908525-84957800-a427-11eb-850e-4c67b4ee236f.png)

**Environment (please complete the following information) **
- OS: MacOS
- Architecture: x86_64
- Shell zsh
- wasmcloud version: 0.18.0

**Additional context**
HELPPP

```[2021-05-10T21:00:24Z INFO ] Starting provider VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4AYJ5RKAX2UNSCAPHA5M
[2021-05-10T21:00:24Z INFO ] Native provider host started
[2021-05-10T21:00:24Z DEBUG] Caching provider to /var/folders/6g/_pz7ywx546l5s2q_d96qb4s00000gn/T/wasmcloudcache/VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4AYJ5RKAX2UNSCAPHA5M/7/x86_64-macos
[2021-05-10T21:00:24Z INFO  wasmcloud_httpserver] Dispatcher configured.
[2021-05-10T21:00:24Z INFO ] Creating lattice subscription for wasmbus://wasmcloud/httpserver/default/VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4AYJ5RKAX2UNSCAPHA5M
[2021-05-10T21:00:24Z INFO ] Native Capability Provider 'wasmbus://wasmcloud/httpserver/default/VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4AYJ5RKAX2UNSCAPHA5M' ready
[2021-05-10T21:00:24Z DEBUG] Advertising MBG3V5EZID5E54ERAM7LB4GE7Z7Y6QTLC5DODKZQD24ESNISCVBNLOMA:default:VAG3QITQQ2ODAOWB5TTQSDJ53XK3SHBEIFNK4AYJ5RKAX2UNSCAPHA5M
[2021-05-10T21:00:24Z INFO  wasmcloud_httpserver] Received HTTP Server configuration for MBG3V5EZID5E54ERAM7LB4GE7Z7Y6QTLC5DODKZQD24ESNISCVBNLOMA
[2021-05-10T21:00:24Z DEBUG wasmcloud_httpserver] HttpServer configured for 0.0.0.0:8080
[2021-05-10T21:00:24Z INFO  actix_server::builder] Starting 8 workers
[2021-05-10T21:00:24Z INFO  actix_server::builder] Starting "actix-web-service-0.0.0.0:8080" service on 0.0.0.0:8080
[2021-05-10T21:00:37Z ERROR] Error invoking actor: Guest call failure: missing imported function (from wasmbus://MBG3V5EZID5E54ERAM7LB4GE7Z7Y6QTLC5DODKZQD24ESNISCVBNLOMA/HandleRequest)
[2021-05-10T21:00:37Z INFO  actix_web::middleware::logger] 127.0.0.1:56986 "GET / HTTP/1.1" 500 29 "-" "curl/7.64.1" 0.000864
make: *** [start] Bus error: 10```

--- a/crates/wasmcloud-host/Cargo.toml
+++ b/crates/wasmcloud-host/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "wasmcloud-host"
-version = "0.18.1"
+version = "0.18.2"
 authors = ["wasmCloud Team"]
 edition = "2018"
 homepage = "https://wasmcloud.dev"

--- a/crates/wasmcloud-host/src/dispatch.rs
+++ b/crates/wasmcloud-host/src/dispatch.rs
@@ -66,8 +66,16 @@ impl Dispatcher for ProviderDispatcher {
             op,
             msg.to_vec(),
         );
-        match block_on(async { self.addr.send(inv).await.map(|ir| ir.msg) }) {
-            Ok(v) => Ok(v),
+        match block_on(async { self.addr.send(inv).await }) {
+            Ok(ir) => {
+                if let Some(e) = ir.error {
+                    let f = format!("Invocation failure: {}", e);
+                    error!("{}", f);
+                    Err(f.into())
+                } else {
+                    Ok(ir.msg)
+                }
+            }
             Err(_e) => {
                 error!("Provider dispatch to bus failed (mailbox error)");
                 Err("Mailbox error during provider dispatch".into())

